pkg/registry/apis/secret/accesscontrol.go

```diff
@@ -0,0 +1,225 @@
+			return authorizer.DecisionDeny, "user required", err
+		}
+
+		// TODO: is this correct?
+		// If the namespace the resource's in differs from the namespace of the user identity, deny.
+		if namespaceInfo.OrgID != user.GetOrgID() {
+			return authorizer.DecisionDeny, "mismatched org", nil
+		}
+
+		// Based on the `kind` being requested, we use different evaluators with scopes and actions.
+		var evaluatorForVerb map[string]accesscontrol.Evaluator
+
+		switch resource {
+		case resourceSecureValues:
+			// TODO: Scope the permission based on the resource name for `Update`, `Read` and `Delete`.
+			// Currently the scope granted for all actions are '*' (all), within the boundaries of a `namespace`.
+			// This would allow fine-grained secret access with RBAC, but we need some way of granting it.
+			scope := ScopeProviderSecretsManagerSecureValues.GetResourceScopeName(name)
+
+			evaluatorForVerb = map[string]accesscontrol.Evaluator{
+				utils.VerbCreate: accesscontrol.EvalPermission(ActionSecretsManagerSecureValuesWrite),
+				utils.VerbGet:    accesscontrol.EvalPermission(ActionSecretsManagerSecureValuesDescribe, scope),
+				utils.VerbList:   accesscontrol.EvalPermission(ActionSecretsManagerSecureValuesList),
+				utils.VerbUpdate: accesscontrol.EvalPermission(ActionSecretsManagerSecureValuesWrite, scope),
+				utils.VerbDelete: accesscontrol.EvalPermission(ActionSecretsManagerSecureValuesDelete, scope),
+			}
+
+		case resourceKeepers:
+			// TODO: Scope the permission based on the resource name for `Update`, `Read` and `Delete`.
+			// Currently the scope granted for all actions are '*' (all), within the boundaries of a `namespace`.
+			// This would allow fine-grained secret access with RBAC, but we need some way of granting it.
+			scope := ScopeProviderSecretsManagerKeepers.GetResourceScopeName(name)
+
+			evaluatorForVerb = map[string]accesscontrol.Evaluator{
+				utils.VerbCreate: accesscontrol.EvalPermission(ActionSecretsManagerKeepersWrite),
+				utils.VerbGet:    accesscontrol.EvalPermission(ActionSecretsManagerKeepersDescribe, scope),
+				utils.VerbList:   accesscontrol.EvalPermission(ActionSecretsManagerKeepersList),
+				utils.VerbUpdate: accesscontrol.EvalPermission(ActionSecretsManagerKeepersWrite, scope),
+				utils.VerbDelete: accesscontrol.EvalPermission(ActionSecretsManagerKeepersDelete, scope),
+			}
+
+		default:
+			return authorizer.DecisionDeny, "unknown resource: " + resource, nil
+		}
+
+		// TODO: when using the LIST verb, how can we filter by only those they have access to?! Need to do it in storage?
+		// TODO: when using the CREATE verb, how can we restrict the scope as well?
+		evaluator, ok := evaluatorForVerb[verb]
+		if !ok {
+			return authorizer.DecisionDeny, "forbidden action: " + verb, nil
+		}
+
+		allowed, err := accessControl.Evaluate(ctx, user, evaluator)
+		if allowed {
+			return authorizer.DecisionAllow, "", nil
+		}
+
+		return authorizer.DecisionDeny, "insufficient permissions", err
+	}
+}
```

Comment:
```suggestion
func SecretAuthorizer(authzc types.AccessClient) authorizer.AuthorizerFunc {
	return func(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
		u, ok := types.AuthInfoFrom(ctx)
		if !ok {
			return authorizer.DecisionDeny, "valid user is required", errors.New("valid user required")
		}

		cr, err := authzc.Check(ctx, u, types.CheckRequest{
			Verb:        attr.GetVerb(),
			Group:       attr.GetAPIGroup(),
			Resource:    attr.GetResource(),
			Namespace:   attr.GetNamespace(),
			Name:        attr.GetName(),
			Subresource: attr.GetSubresource(),
		})

		if err != nil {
			return authorizer.DecisionDeny, "authorization failed", err
		}

		if cr.Allowed {
			return authorizer.DecisionAllow, "", nil
		} else {
			return authorizer.DecisionDeny, "denied", nil
		}
	}
}
```

For our new apis we should use `types.AccessClient` (injected by wire). This client also supports multi-tenant access checks. 

The only thing you would need to do then is to setup mapping between rbac actions and k8s style checks here https://github.com/grafana/grafana/blob/main/pkg/services/authz/rbac/mapper.go#L55-L67

Reply:
Or better yet just use https://github.com/grafana/grafana/blob/main/pkg/services/apiserver/auth/authorizer/resource.go

Reply:
Amazing 🎉 I used both and now we don't need to maintain our poor version of an authorizer 😄